Add localization for French

mike011 opened the pull request:
## What was done?
* Added Localizable.string and localized xibs.

## Why?
https://github.com/pedrommcarrasco/Brooklyn/issues/58

## Screenshots
If there's any relevant screenshot please attach it here.
<img width="983" alt="Screen Shot 2019-10-07 at 9 15 25 PM" src="https://user-images.githubusercontent.com/705433/66360074-ae6f7280-e947-11e9-8c22-06e628f859dc.png">

pedrommcarrasco commented:
Please pull from master, a PR was merged yesterday

pedrommcarrasco commented on Brooklyn/Contexts/Preferences/fr.lproj/PreferencesWindowController.strings:
```diff
@@ -0,0 +1,72 @@
+
+/* Class = "NSTextFieldCell"; title = "Want to keep updated?"; ObjectID = "3ad-ri-H5Y"; */
+"3ad-ri-H5Y.title" = "Voulez-vous rester à jour?";
```
Not a huge fan of Localizable xibs as they will spread our translations across multiple files. Can we have them all in `Localizable.strings`?

mike011 replied:
I've removed the xib location files and converted all references to using Localizable.strings.

pedrommcarrasco replied:
Yeah I noticed, the only thing "missing" is this:
https://github.com/pedrommcarrasco/Brooklyn/pull/60#discussion_r332912689

pedrommcarrasco commented on Brooklyn/Resources/fr.lproj/Localizable.strings:
```diff
@@ -0,0 +1,70 @@
+*/
+
+"Apple's Original Version" = "Version originale d'Apple";
+"Avatar Tissue"= "Tissu Avatar";
```
Can we make the keys a little more formatted?
For example: `animation.appleOriginalVersion`

mike011 replied:
Great idea, I found a much simpler way to do this. Hope you like it :) In Animation use the enum value as the key and look that up in the Localizable.strings files. It makes the Animation enum super simple :)

mike011 replied:
I pushed up the change just for the Animations will do the menus and the rest of the string when I find a good solution to deal with xib based localizations.

pedrommcarrasco replied:
> do the menus and the rest of the string when I find a good solution to deal with xib based localizations.

Regarding this, move them to this file with good key names. I'm not usual a XIB user (I tend to do things by code, except in Brooklyn 😅) but I would assume you'll have to associate localizable strings to an UI element in code.

pedrommcarrasco replied:
https://medium.com/@mario.negro.martin/easy-xib-and-storyboard-localization-b2794c69c9db This doesn't look bad either

mike011 replied:
Great suggestion. I've implemented and agree it's much cleaner and long term maintainable :)

pedrommcarrasco commented:
Thanks for taking the initiative and joining us @mike011! Or should I say, merci? :D

pedrommcarrasco commented on Brooklyn/Resources/Localization/en.lproj/Localizable.strings:
```diff
@@ -0,0 +1,70 @@
+  Copyright © 2019 Pedro Carrasco. All rights reserved.
+*/
+
+"original" = "Apple's Original Version";
```
Still not my cup of tea as it doesn't provide context.
You could add the prefix ´animation´ -> ´animation.original´.
Then it would become:
```swift
extension Animation {

    var name: String {
        return "animation.\(rawValue)".localized
    }
}
```

What do you think?

pedrommcarrasco replied:
Did you forget this? Also, the rest of the translations should also have a prefix in my opinion, such as ´menu.` and `preference.`

mike011 replied:
I agree for animation that is makes sense to have it for extra information. But for menu and preference I feel normally that's it too specific. At the localizable level you normally want to be able to reuse the strings. But in this case the strings seem pretty unique per set. So I've gone ahead and added the changes.

pedrommcarrasco replied:
I kinda of prefer being specific and then generalise if needed 👍 
Thanks for everything @mike011 and welcome to the list of contributors! ❤️

mike011 commented:
Please review @pedrommcarrasco :)

pedrommcarrasco commented:
I've left some comments 5 days ago @mike011 😄

mike011 commented:
@pedrommcarrasco My bad. I addressed the feedback, but didn't say I addressed the feedback. Here's what I did. I've added `XIBLocalizable.swift` which adds a user defined attribute called `xibLocKey` to `NSTextFieldCell`, `NSButtonCell`, `NSWindow`, `NSMenu`, and `NSMenuItem`. I then went through and added the appropriate string into `xibLocKey` for all the cells, windows, menus, and menu items.

pedrommcarrasco commented:
Yeah I saw that, the only thing left is these two comments 👇
https://github.com/pedrommcarrasco/Brooklyn/pull/60#discussion_r332912689

Also, let me know if you're running out of time, I can do that myself later if needed!
